inc/class-config.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Core platform config loader.
+ *
+ * @package hm-platform-core
+ */
+
+namespace HM\Platform;
+
+class Config {
+
+	/**
+	 * Primary config array.
+	 *
+	 * @var array
+	 */
+	protected static $config = [];
+
+	/**
+	 * Retrieve the configuration for HM Platform.
+	 *
+	 * The configuration is defined by merging the defaults set by modules
+	 * with any overrides present in composer.json.
+	 *
+	 * @return array Configuration data.
+	 */
+	static function get() : array {
+		if ( self::$config ) {
+			return self::$config;
+		}
+
+		self::$config = self::get_merged_defaults_and_customizations();
+
+		return self::$config;
+	}
+
+	/**
+	 * The environment type is used to override the config
+	 * as well as allow for checks within a project codebase.
+	 *
+	 * Defaults to local, this constant will be set at the stack level.
+	 *
+	 * @return string The environment type string.
+	 */
+	static function get_environment_type() : string {
```

Comment:
I think we might need to generally fix this pattern of reliance on functions. For anyone else: this has been copied from `inc/namespace.php` because there's no guerentee that function would be available yet here.

Either we need to accept `Config::get()` can only be used after the autoloader has run (defeating the purpose of using a class for Config`, or we require no functions be used from elsewhere.

Any ideas @rmccue ?

Reply:
It's probably a bit of an anti-pattern but the helper functions could be included in the `Config::get()` method, or included at the top of the config class file. Both of those options are a bit gross. Would likely need to include the helpers file in the modules object too so it's available in module bootstrap code.

Reply:
Resolved, will be using `load.php` and actions/filters.